File: account/src/main/java/serp/project/account/core/domain/dto/request/BulkAssignUsersRequest.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.core.domain.dto.request;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class BulkAssignUsersRequest {

    @NotNull(message = "Organization ID is required")
    private Long organizationId;

    @NotNull(message = "Module ID is required")
    private Long moduleId;

    @NotEmpty(message = "User IDs list cannot be empty")
    private List<Long> userIds;
}

File: api_gateway/src/core/domain/dto/request/account/bulk_assign_users_request.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type BulkAssignUsersRequest struct {
	OrganizationId int64   `json:"organizationId"`
	UserIds        []int64 `json:"userIds"`
	ModuleId       int64   `json:"moduleId"`
}
